app/src/main/kotlin/com/github/funkyg/funkytunes/activities/PlayingQueueActivity.kt

```diff
@@ -48,8 +56,80 @@ class PlayingQueueActivity : BaseActivity(), PlaybackInterface {
         finish()
     }
 
+    override fun onProgress(indexInTorrent: Int, progress: Int) {
+        val playlist: List<Song>? = currentPlaylist
+        if (playlist == null) {
+            Log.v(Tag, "Progress received when playlist is null")
+            return
+        }
+
+        val songs = playlist.filter({s -> s.indexInTorrent == indexInTorrent})
+        if (songs.size == 0) {
+            Log.v(Tag, "Progress received for unknown file index $indexInTorrent")
+        }
+        val song = songs[0]
+        val lastProgress = song.progress
+        if (!song.isPlaying) {
+            song.isPlaying = false
+            song.progress = progress.toInt()
+
+            //Log.v(Tag, "Progress[$index] = $progress song=$song")
+            if(lastProgress != song.progress) {
+                runOnUiThread {
+                    binding.recycler.getAdapter().notifyDataSetChanged()
```

Comment:
i guess you can ignore this, it shouldnt be a problem once the progressbar works correctly